src/frontend/src/optimizer/plan_expr_visitor/strong.rs

```diff
@@ -0,0 +1,306 @@
+    }
+
+    fn is_null_visit(&self, expr: &ExprImpl) -> bool {
+        match expr {
+            ExprImpl::InputRef(input_ref) => self.is_input_ref_null(input_ref),
+            ExprImpl::Literal(literal) => literal.get_data().is_none(),
+            ExprImpl::FunctionCall(func_call) => self.is_null_function_call(func_call),
+            ExprImpl::FunctionCallWithLambda(_) => false,
+            ExprImpl::AggCall(_) => false,
+            ExprImpl::Subquery(_) => false,
+            ExprImpl::CorrelatedInputRef(_) => false,
+            ExprImpl::TableFunction(_) => false,
+            ExprImpl::WindowFunction(_) => false,
+            ExprImpl::UserDefinedFunction(_) => false,
+            ExprImpl::Parameter(_) => false,
+            ExprImpl::Now(_) => false,
+        }
+    }
+
+    fn is_null_function_call(&self, func_call: &FunctionCall) -> bool {
+        match func_call.func_type() {
+            // NOT NULL: This kind of expression is never null. No need to look at its arguments, if it has any.
+            ExprType::IsNull
+            | ExprType::IsNotNull
+            | ExprType::IsDistinctFrom
+            | ExprType::IsNotDistinctFrom
+            | ExprType::IsTrue
+            | ExprType::IsNotTrue
+            | ExprType::IsFalse
+            | ExprType::IsNotFalse => false,
+            // ANY: This kind of expression is null if and only if at least one of its arguments is null.
+            ExprType::Not
+            | ExprType::Equal
+            | ExprType::NotEqual
+            | ExprType::LessThan
+            | ExprType::LessThanOrEqual
+            | ExprType::GreaterThan
+            | ExprType::GreaterThanOrEqual
+            | ExprType::Like
+            | ExprType::Add
+            | ExprType::AddWithTimeZone
+            | ExprType::Subtract
+            | ExprType::Multiply
+            | ExprType::Modulus
+            | ExprType::Divide
+            | ExprType::Cast
+            | ExprType::Trim
+            | ExprType::Ltrim
+            | ExprType::Rtrim
+            | ExprType::Ceil
+            | ExprType::Floor
+            | ExprType::Extract
+            | ExprType::Greatest
+            | ExprType::Least => self.any_null(func_call),
+            ExprType::And | ExprType::Or | ExprType::Coalesce => self.all_null(func_call),
+            // TODO: Function like case when is important but current its structure is complicated, so we need to implement it later if necessary.
+            // Assume that any other expressions cannot be simplified.
+            _ => false,
+        }
+    }
+
+    fn any_null(&self, func_call: &FunctionCall) -> bool {
+        func_call
+            .inputs()
+            .iter()
+            .any(|expr| self.is_null_visit(expr))
+    }
+
+    fn all_null(&self, func_call: &FunctionCall) -> bool {
+        func_call
+            .inputs()
+            .iter()
+            .all(|expr| self.is_null_visit(expr))
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Could we have some nested / complicated expressions tests here just for reference? (Also some corner case expression, e.g., [`false and null` / `true or null`] / [`false or null` / `true and null`]

Reply:
All these cases should be handled by constant folding, before using `is_null` to test them if we want to accurately analyze their property.

Reply:
> nested / complicated expressions
What about these? 👀